pkg/3scale/amp/operator/system_reconciler.go

```diff
@@ -246,14 +246,24 @@ func (r *SystemReconciler) validateS3StorageProvidedConfiguration() error {
 
 	secretData := awsSecret.Data
 	var result *string
-	result = helper.GetSecretDataValue(secretData, component.AwsAccessKeyID)
-	if result == nil {
-		return fmt.Errorf("Secret field '%s' is required in secret '%s'", component.AwsAccessKeyID, awsCredentialsSecretName)
-	}
-
-	result = helper.GetSecretDataValue(secretData, component.AwsSecretAccessKey)
-	if result == nil {
-		return fmt.Errorf("Secret field '%s' is required in secret '%s'", component.AwsSecretAccessKey, awsCredentialsSecretName)
+	if system.Options.S3FileStorageOptions != nil && system.Options.S3FileStorageOptions.S3STSEnabled {
```

Comment:
`if system.Options.S3FileStorageOptions.S3STSEnabled` is enough as the context says the user has specified S3

Reply:
done. thank you